content/docs/command-reference/add.md

```diff
@@ -70,6 +72,21 @@ large files. DVC also supports other link types for use on file systems without
 `reflink` support, but they have to be specified manually. Refer to the
 `cache.type` config option in `dvc config cache` for more information.
 
+### Transferring data directly to the remote
+
+Giving `--to-remote` option would change the behavior described above. Instead
+of only being able to give it something from local/remote workspace, it would be
+able to support all kinds of remote locations that you can import something
+(listed in [import-url](/doc/command-reference/import-url)). The main difference
+is that it won't actually do anything on the workspace beside creating a DVC
+file. It will take the data in batches from the given target and transfer it
```

Comment:
I think `in batches` and `through the local system` are implementation details. We can make them a note if they are important details for the end user. But otherwise it's better to keep this a bit higher level and omit such details.